pkg/querier/seriesset.go

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2018 Iguazio Systems Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License") with
+an addition restriction as set forth herein. You may not use this
+file except in compliance with the License. You may obtain a copy of
+the License at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+In addition, you may not use the software for any purposes that are
+illegal under applicable law, and the grant of the foregoing license
+under the Apache 2.0 license is conditioned upon your compliance with
+such restriction.
+*/
+
+package querier
+
+import (
+	"github.com/nuclio/logger"
+	"github.com/v3io/v3io-go-http"
+	"github.com/v3io/v3io-tsdb/pkg/aggregate"
+	"github.com/v3io/v3io-tsdb/pkg/partmgr"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+)
+
+func newSeriesSet(partition *partmgr.DBPartition, mint, maxt int64) *V3ioSeriesSet {
+
+	return &V3ioSeriesSet{mint: mint, maxt: maxt, partition: partition}
+}
+
+// holds the query result set
+type V3ioSeriesSet struct {
+	err        error
+	logger     logger.Logger
+	partition  *partmgr.DBPartition
+	iter       utils.ItemsCursor
+	mint, maxt int64
+	attrs      []string
+	chunkIds   []int
+
+	interval   int64
+	nullSeries bool
+	overlapWin []int
+	aggrSeries *aggregate.AggregateSeries
+	aggrIdx    int
+	currSeries Series
+	aggrSet    *aggregate.AggregateSet
+	baseTime   int64
+}
+
+// Get relevant items & attributes from the DB, and create an iterator
+// TODO: get items per partition + merge, per partition calc attrs
+func (s *V3ioSeriesSet) getItems(partition *partmgr.DBPartition, name, filter string, container *v3io.Container, workers int) error {
+
+	path := partition.GetTablePath()
+	shardingKeys := []string{}
+	if name != "" {
+		shardingKeys = partition.GetShardingKeys(name)
+	}
+	attrs := []string{"_lset", "_ooo", "_name", "_maxtime"}
```

Comment:
What `_ooo` stands for?

Reply:
@imakhlin out of order

Reply:
ok, but it has not been used yet.
I would suggest more human readable name, maybe `_in_order` = [true|false] (or `_sorted`)